main.go

```diff
@@ -100,3 +116,65 @@ func renderPage(page string) string {
 	}
 	return result
 }
+
+type pageCache struct {
+	diskv *diskv.Diskv
+}
+
+// CachedPage is a page stored in the diskv cache.
+type CachedPage struct {
+	// The page content.
+	Page string
+	// Unix time (seconds).
+	CreatedAt int64
+}
+
+func (c *CachedPage) isValid(now int64) bool {
+	return now-c.CreatedAt < 60*60*24 // 1 day.
+}
+
+func newCache() *pageCache {
+	currentUser, err := user.Current()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not read home directory: %v\n", err)
+		os.Exit(-1)
+	}
+
+	return &pageCache{
+		diskv: diskv.New(diskv.Options{
+			BasePath: filepath.Join(currentUser.HomeDir, ".tldr", "cache"),
+		})}
+}
+
+func (c *pageCache) write(page string, content string) {
+	var cacheBuffer bytes.Buffer
+	encoder := gob.NewEncoder(&cacheBuffer)
+	encoder.Encode(CachedPage{
+		CreatedAt: time.Now().Unix(),
+		Page:      content,
+	})
+	c.diskv.Write(page, cacheBuffer.Bytes())
+}
+
+func (c *pageCache) read(page string) (string, error) {
+	var cachedData []byte
+	var err error
+	if cachedData, err = c.diskv.Read(page); err != nil {
+		return "", errors.New("Error loading from cache")
+	}
+
+	decoder := gob.NewDecoder(strings.NewReader(string(cachedData)))
+	cp := CachedPage{}
+	if err = decoder.Decode(&cp); err != nil {
+		// Ups, cached data is corrupted.
+		c.diskv.Erase(page)
```

Comment:
honestly it does the right thing, but it seems an overly smart cache that has a write side effect on a read() operation. Given the chances of data corruption are so low, perhaps just err out and quit?